gryphon/lib/exchange/binance_je.py

```diff
@@ -0,0 +1,188 @@
+        request header and a timestamp and signature within the query string.
+        """
+        self.load_credentials()
+
+        try:
+            headers = request_args["headers"]
+        except KeyError:
+            headers = request_args["headers"] = {}
+
+        headers["X-MBX-APIKEY"] = self.credentials["api_key"]
+
+        try:
+            params = request_args["params"]
+        except KeyError:
+            params = request_args["params"] = {}
+
+        timestamp = int(round(time.time() * 1000))
+        params["timestamp"] = timestamp
+
+        signature = binance_signature(params, self.credentials["secret"])
+        params["signature"] = signature
+
+        logger.debug("headers: %s" % headers)
+        logger.debug("request args: %s" % request_args)
+
+    def ping(self):
+        req = self.req(no_auth=True, **self.endpoints["ping"])
+        return self.resp(req)
+
+    def get_balance_req(self):
+        return self.req(**self.endpoints["balance"])
+
+    def get_balance_resp(self, req):
+        response = self.resp(req)
+        logger.debug("balance repsonse: %s" % response)
+
+        try:
+            balances = {
+                currency: Money(balance["free"], currency)
+                for balance in response["balances"]
+                for currency in Money.CURRENCIES
+                if balance["asset"] == currency
+            }
+            logger.debug("balances: %s" % balances)
+        except KeyError:
+            raise exceptions.ExchangeAPIErrorException(
+                self, "Cannot determine balances from response"
+            )
+
+        return Balance(balances)
+
+    def get_prices(self):
+        req = self.get_prices_req()
+        return self.get_prices_resp(req)
+
+    def get_prices_req(self):
+        return self.req(no_auth=True, **self.endpoints["prices"])
+
+    def get_prices_resp(self, req):
+        response = self.resp(req)
+        logger.debug("prices response: %s" % response)
+        return {
+            item["symbol"][:3]: Money(item["price"], item["symbol"][-3:])
+            for item in response
+            if item["symbol"][:3] in Money.CRYPTO_CURRENCIES
+        }
+
+    def get_open_orders_req(self):
+        return self.req(**self.endpoints["open_orders"])
+
+    def get_open_orders_resp(self, req):
+        response = self.resp(req)
+        logger.debug("open orders response: %s" % response)
+        side_to_mode = {"BUY": Consts.BID, "SELL": Consts.ASK}
+        return [
+            {
+                "mode": side_to_mode[order["side"]],
+                "id": order["orderId"],
+                "price": order["price"],
+                "volume_remaining": Money(
+                    order["origQty"] - response["executedQty"], self.volume_currency
+                ),
+            }
+            for order in response
+        ]
+
+
+class BinanceJeBTCEURExchange(BinanceJeExchange):
+    currencies = {"volume": "BTC", "base": "EUR"}
```

Comment:
Gryphon relies on the parameters `self.volume_currency` and `self.price_currency` being set in the constructor, so I'd replace this dictionary by setting those parameters.

Reply:
OK. Will do.

Reply:
I set those in the __init__ usig this dict (although, I used self.currency as I copied it from one of the other implementations)

Reply:
L60 and L61

Reply:
I've left this as is but attempted to explain it in the docstring for the class.